examples/main.tf

```diff
@@ -0,0 +1,74 @@
+provider "azurerm" {
+  version = "1.22.1"
+  subscription_id = "${var.subscription_id}"
+}
+variable "subscription" {}
+
+variable "subscription_id" {}
+
+variable "product" {
+  default = "idam"
+}
+
+variable "location" {
+  default = "UK South"
+}
+
+variable "env" {
+  default = "sandbox"
+}
+
+variable "capacity" {
+  default = "1"
+}
+
+
+variable "common_tags" {
+  default = {
+    "Team Name" = "IDAM"
+  }
+}
+
+# Both default_asp_name_and_rg and params_asp_name_and_rg must be included to 
+# ensure that the dependencies and references work correctly in both examples.
+# A common error is to include a new dependsOn resourceId without the RG
+# parameter, which will work for default_asp_name_and_rg but fail on
+# params_asp_name_and_rg.
+module "default_asp_name_and_rg" {
```

Comment:
not sure what you're trying to achieve here, the second example is what people should be doing by default

Reply:
The second example `params_asp_name_and_rg` is how IDAM is implemented however the module supports no ASP and RG name parameters, which affects the successful execution of the ARM template if the `dependsOn` resource is not referenced using the `resourceId` parameter. 

Example: https://github.com/hmcts/cnp-module-webapp/blob/5d7fe9a1ad743efb4581bc50e99661041576b2ec/templates/asp-app.json#L157

This examples directory simply tests both module configurations for a more complete test.